enclave/chain_relay/src/justification.rs

```diff
@@ -113,12 +71,23 @@ impl<Block: BlockT> GrandpaJustification<Block> {
 			let msg = "invalid commit target in grandpa justification".to_string();
 			Err(ClientError::BadJustification(msg))
 		} else {
-			justification.verify(set_id, voters).map(|_| justification)
+			justification.verify_with_voter_set(set_id, voters).map(|_| justification)
 		}
 	}
 
 	/// Validate the commit and the votes' ancestry proofs.
-	pub(crate) fn verify(
+	pub fn verify(&self, set_id: u64, authorities: AuthorityList) -> Result<(), ClientError>
+	where
+		NumberFor<Block>: finality_grandpa::BlockNumberOps,
+	{
+		let voters =
+			VoterSet::new(authorities.iter().cloned()).ok_or(ClientError::InvalidAuthoritiesSet)?;
```

Comment:
I think `iter().clone()` can be replaced by `into_iter`?

Reply:
Copy pasted from substrate (https://github.com/paritytech/substrate/blob/master/client/finality-grandpa/src/justification.rs#L125) but I'll change accordingly.

Reply:
Resolved